alt_e2eshark/onnx_tests/models/azure_models.py

```diff
@@ -113,3 +134,38 @@ def construct_model(self):
 
 for t in basic_opt:
     register_test(AzureWithOpt, t)
+
+
+class AzureWithOptAndNoOpsetVersion(AzureWithOpt):
+    def __init__(self, name: str, onnx_model_path: str):
+        super().__init__(name, onnx_model_path)
+        self.opset_version = None
+
+
+for t in no_opset_update:
+    register_test(AzureWithOptAndNoOpsetVersion, t)
+
+
+class AzureRemoveMetadataProps(AzureDownloadableModel):
+    def construct_model(self):
+        super().construct_model()
+        self.remove_duplicate_metadata_props()
+
+    def remove_duplicate_metadata_props(self):
+        model = onnx.load(self.model)
+        metadata = model.metadata_props
+        # unhashable type, so manually check for duplicates
+        to_remove = []
+        for i, m0 in enumerate(metadata):
+            for j in range(i + 1, len(metadata)):
+                if m0 == metadata[j]:
+                    to_remove.append(i)
+        num_removed = 0
+        for i in to_remove:
+            _ = metadata.pop(i - num_removed)
+        mod_model_path = str(Path(self.model).parent / "model.modified.onnx")
+        onnx.save(model, mod_model_path)
+        self.model = mod_model_path
+
+
+register_test(AzureRemoveMetadataProps, "resnetv2_50x1_bit.goog_in21k_ft_in1k_vaiq")
```

Comment:
LG. Wondering, what the reason for the duplicate meta data is?

Reply:
No idea. Likely an accident by someone who was quantizing it. Normally not a problem, but the onnx importer uses a model checker and it's failing there.